common/test/src/View/CompilationRunnerTest.cpp

```diff
@@ -130,6 +130,35 @@ TEST_CASE_METHOD(
   CHECK(testEnvironment.directoryExists(targetPath));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "CompilationDeleteFilesTaskRunner.deleteTargetPattern")
+{
+  auto variables = EL::NullVariableStore{};
+  auto output = QTextEdit{};
+  auto outputAdapter = TextOutputAdapter{&output};
+
+  auto context = CompilationContext{document, variables, outputAdapter, false};
+
+  auto testEnvironment = IO::TestEnvironment{};
+
+  const auto file1 = IO::Path("file1.lit");
```

Comment:
You should also test error cases: The file doesn't exist, the file cannot be deleted, the path actually denotes a directory.

Reply:
"File doesn't exist" is a little tricky since the delete spec is just a pattern which may or may not match anything. It also probably should not be an error case anyway to ask to delete a path that has already been deleted... e.g. the usecase for "delete an existing .lit file if there is one" should complete successfully whether or not the .lit file exists.

I do at least check that the pattern doesn't delete other files it's not supposed to delete.

I'll add a check that it also doesn't delete directories even if they match the pattern.

"File cannot be deleted", hmm, I'm not sure about how to test that. I guess this would involve connecting something to the Qt "error" signal that the runner would emit when it encounters an error? The other part of this would be figuring out how to actually trigger the error... I don't think DiskIO (or some other interface) currently supports setting file ACLs does it?

Reply:
Directory check done. I'm not planning any other changes at the moment, pending feedback about the questions above.

Reply:
Yeah, you are right, I didn't think it through. My lizard brain just automatically rattled off anything that could possibly go wrong ;-)